ebs/error_budget.py

```diff
@@ -417,6 +410,407 @@ def run_etc(self, wfe, wfsc_factor, sensitivity
+                indices = np.where(np.isfinite(template))
+                use_values, values = np.split(values, [indices[0].size])
+                arr = getattr(self, var_name)
+                if type(arr) == float or type(arr)==np.float64:
+                    arr = use_values[0]
+                else:
+                    arr[indices] = use_values
+                setattr(self, var_name, arr)
+                if var_name in ['contrast', 'wfe', 'wfsc_factor'
+                                , 'sensitivity']:
+                    if var_name == 'contrast':
+                        self.write_csv(var_name)
+                        self.exosims_pars_dict['starlightSuppressionSystems']\
+                                [0]['core_contrast'] = self.contrast_filename
+                    self.write_ppFact_fits()
+                    self.exosims_pars_dict['ppFact'] = self.ppFact_filename
+                if var_name == 'throughput':
+                    self.write_csv(var_name)
+                    self.exosims_pars_dict['starlightSuppressionSystems']\
+                            [0]['core_thruput'] = self.throughput_filename
+            else:
+                print(key+" not found in attributes list")
+
+    def log_prior(self, values):
+        joint_prob = 0.0
+        counter = 0
+        for i, var_name in enumerate(self.config['mcmc']['variables'].keys()):
+            prior_ftn = np.array(
+                    self.config['mcmc']['variables'][var_name]['prior_ftn']
+                                )
+            index = np.where(prior_ftn!='nan')
+            all_args = [np.array(item) for item in 
+                    self.config['mcmc']['variables'][var_name]['prior_args']
+                        .values()]
+            for m, row_index in enumerate(index[0]):
+                if len(index) > 1:
+                    col_index = index[1][m]
+                    ftn = getattr(pdf, prior_ftn[row_index, col_index])
+                    args = [page[row_index][col_index] for page in all_args]
+                else:
+                    ftn = getattr(pdf, prior_ftn[row_index])
+                    args = [page[row_index] for page in all_args]
+                prob = ftn(values[counter], *args)
+                counter += 1
+                if np.isinf(prob):
+                    return prob
+                else: 
+                    joint_prob += prob
+        return joint_prob 
+
+    def log_merit(self, values):
+        self.update_attributes(values)
+        int_time = self.run_exosims()[0]
+        if np.isnan(int_time.value).any():
+            return -np.inf
+        else:
+            ftn_name = self.config['mcmc']['merit']['ftn']
+            args = [arg for arg in self.config['mcmc']['merit']['args']\
+                    .values()]
+            ftn = getattr(pdf, ftn_name)
+            mean_int_time = np.array(int_time.value).mean() 
+            return ftn(mean_int_time, *args)
+
+#    def log_probability(self, values):
```

Comment:
I would just remove this comment since the code is preserved elsewhere in the file